src/segment_session.go

```diff
@@ -29,12 +31,26 @@ const (
 	SSHIcon Property = "ssh_icon"
 	// DefaultUserName holds the default user of the platform
 	DefaultUserName Property = "default_user_name"
+
+	defaultUserEnvVar = "POSH_SESSION_DEFAULT_USER"
 )
 
 func (s *session) enabled() bool {
 	s.UserName = s.getUserName()
+	s.ComputerName = s.getComputerName()
+	s.SSHSession = s.activeSSHSession()
+	s.DefaultUserName = s.getDefaultUser()
+	segmentTemplate := s.props.getString(SegmentTemplate, "")
+	if segmentTemplate != "" {
+		s.Root = s.env.isRunningAsRoot()
+		template := &textTemplate{
+			Template: segmentTemplate,
+			Context:  s,
+		}
+		s.templateText = template.render()
+		return len(s.templateText) > 0
```

Comment:
😆 Logic bug, so this broke the current behavior as it returns too early and is always enabled.
Since `getUsername` always returns the username so the segmentTemplate is always greater than 0.

I tried fixing the code to have `segmentTemplate` as in have `getUsername` should return `""`\
when `showDefaultUser == false && defaultUserName == username`.

Sadly that breaks a lot of tests 😅

So the enabled would look something like this.
```go
	if segmentTemplate != "" {
		s.Root = s.env.isRunningAsRoot()
		template := &textTemplate{
			Template: segmentTemplate,
			Context:  s,
		}
		s.templateText = template.render()
		return len(s.templateText) > 0
	}
	return false
```

Not sure where the logic for `showDefaultUser` should be located as in you would want to show host or session host depending on other logic. 😓